### ISSUE TITLE
Contacting about AUR package maintain

### ISSUE THREAD

**Issue by Gigas002** [dr460nf1r3/dr460nf1r3#1]
Hey, @dr460nf1r3, hope you're doing great! Sorry to contact you here, didn't find a better place on github to contact you :)
I'm the maintainer of `plasma5-themes-sweet-full-git` and `plasma5-themes-sweet-kde-git` packages in AUR.
I'm currently tracking the state of KDE 6 update on theme repo, and was thinking about renaming in the future, since theme would _probably_ work on both KDE 5 and 6. During this I've noticed that `garuda` is probably using my packages for theming, since you've left a comment there and there are some topics on garuda forum, so before changing anything I decided to contact you directly, so I don't break anything important :D
I can also add you as a co-maintainer for these packages, if you think it will help you maintaining this theme on Garuda Linux.

**Comment by dr460nf1r3** [comment]
Hey :) 
Thanks for being as mindful, that's very appreciated! In fact we do use the package, only slightly modified via our build tools to work around breaking renaming of themes. You can see the list of changes here: 

https://github.com/chaotic-aur/interfere/tree/master/plasma5-themes-sweet-full-git

Iirc the GTK theme renaming was added after we switched from the standalone variant to this one, to preserve existing theme configs. 

I don't know whether these would be acceptable for you to include? 

Concerning P6 compatibility: there are several changes outstanding that fix some bugs happening on P6 only. The SDDM theme was also updated with correct qt imports. So yeah, they might not be backwards compatible. The theme itself works on there quite will though apart from the mentioned minor glitches.

https://github.com/EliverLara/Sweet/issues/259

I'm up for co-maintainership, though I'd love to discuss whether the changes above would be feasible.

**Comment by Gigas002** [comment]
Hmm, while I don't have any objections to it, I'd like to not introduce breaking changes if possible. At least for now. If or when we decide on package renaming for p6 migration, I think it'd be better to do at the same time. Would this be OK?

Regarding p6 migration, current package is a `git` package, so it points to a `nova` branch, and I think p6 patch will eventually be merged into it, breaking functionality for p5 users. Plus, current naming won’t correctly represent theme’s behavior.
As a solution, we could point package on a specific commit and freeze forever for p5 users and create a new package for p6, which would act as current one does. I don’t know what would be a more appropriate solution though...

I've seen you also maintain `beautyline` package with icons, maybe we should add it as an optional dependency for a theme?

Aaaanyways, I have added you as a co-maintainer for a package ;)
Feel free to contact/ping me any time to discuss changes.

BTW, FYI I’m currently working on porting partitions of this theme to hyprland via hyprdots. You can check out [this repo](https://github.com/Gigas002/hyprdots-sweet) if you're interested, maybe the Garuda users could benefit from it too. I’ve been using it for nearly 2-3 months and it works fine for me, but I guess it doesn’t yet meets everyone needs

**Comment by dr460nf1r3** [comment]
In case you weren't aware of it - the P6 specific changes already arrived in the latest commits, so everyone building this package as of now automatically gets the P6 stuff. Which means, that the pkgname is a little misleading as of now. My proposal would be creating a `plasma6-themes-sweet-full-git` package and setting up a package merge request via AUR requests. Imo there is little reason for keeping the P5 version around as no one using Arch would have a reason to use it? After all the repo keeps rolling 😁 If we decide to adopt the name changes I mentioned, the pkgname change would indeed be the best time to do it. If we don't, that's also fine. Our build tools can handle that just fine as shown above :) 

As for adding `beautyline` to opt depends, that's a fantastic idea. In my personal opinion, these suit the general appearance much better than the Candy icons (which the `beautyline` icon package also makes use of for missing icons however! It's actually a mix of both packages for maximum consistency). 

Thanks for adding me as a maintainer! :) 

About the hyprdots - I'll show this to our Hyprland edition maintainers. The screenshots look amazing. Not sure whether it fits their personal tast ,but even if not, maybe there is something useful people can learn from there 😄

**Comment by Gigas002** [comment]
> the P6 specific changes already arrived in the latest commits, so everyone building this package as of now automatically gets the P6 stuff.

I'm not sure about that. If you're speaking about latest changes in [sweet-kde](https://github.com/EliverLara/Sweet-kde) repo, then I guess you're right: commits in `master` and `plasma6` branches seems the same. But in case of some other KDE components from [sweet](https://github.com/EliverLara/Sweet) repo that's not yet correct. Right now we're building the theme from [nova branch](https://github.com/EliverLara/Sweet/commits/nova/), which doesn't include any specific p6 migration commits. The notable changes for SDDM were introduced in [dark-plasma-6](https://github.com/EliverLara/Sweet/commits/dark-plasma-6) branch, and I'm guess this as incomplete, until the issue you've mentioned in upstream repo is resolved and everyone's happy.
I wonder though, if plasma 6 branches will be merged into `master`/`nova` branches respectively for each repository, or will be developed separately, and we'll need to change our target branches...

> Imo there is little reason for keeping the P5 version around as no one using Arch would have a reason to use it? After all the repo keeps rolling 😁

I agree. While I think there *might* be some users, that want to stay on p5, like there are some who uses p3 and p4 nowadays, but I doubt they're using arch as their distro.

> `plasma6-themes-sweet-full-git`, pkgname change, add `beautyline` optdep

These changes looks good for me 👍  We'll also need to rename a `plasma5-themes-sweet-kde-git` too, I guess.

> About the hyprdots - I'll show this to our Hyprland edition maintainers. The screenshots look amazing. Not sure whether it fits their personal tast ,but even if not, maybe there is something useful people can learn from there 😄

The `hyprdots` project infrastructure is a bit tricky, and it lacks some conditional flow atm, though I'm suggesting a PR for that. I think it's a good choice for a beginner `hyprland` users to keep their hand on this wayland compositor and start tweaking their configs and repos little by little.
But in any way, I think it's indeed may be useful to at least take a look at how things are done out there, even if it not fits current needs.
Feel free to ping me or file issues about theme for a `hyprland` too, will try to contribute/implement any requests as much as I can :)

**Comment by Gigas002** [comment]
Hey, @dr460nf1r3! Sorry to contact you soon after previous time, I hope you're not too busy lately.
After giving lots of thoughts since our last conversation, I've decided to create a community repo for Sweet themes for different applications, so everyone will be able to unify how their system looks. Garuda users would benefit from it too, I think, though there's not much in there atm. Any contributions are welcomed! I know that sound silly, but I hope open source community would like to contribute with their apps too...
I've already added stuff I created  and currently use in hyprland with some instructions, check [this repo out](https://github.com/Gigas002/awesome-sweet) if you're interested!
I'm also mirroring the PKGBUILDs out there, so maybe discussing AUR packages related questions would be better to do there from now on...
Sorry for being annoying once again.
That's kinda it. Have a good day!

**Comment by dr460nf1r3** [comment]
You are right about the KDE 6 components not being fully included in the nova branch, now that I looked into it. Sorry for the confusion. 
That repo you created looks neat! Don't have the time to look into it and answer in-depth, but will come back at it at a later time. Thanks for letting me know! :)

**Comment by kenderipa** [comment]
Hey @dr460nf1r3, I also have a requst regarding a Chaotic AUR package. I'm trying to build a VCMI AppImage, and with the help of https://github.com/pkgforge-dev/Anylinux-AppImages/blob/main/useful-tools/quick-sharun.sh, it's creator Samueru Sama and Chaotic AUR build of VCMI I've almost succeeded. Everything works, except importing the data for playing Heroes Chronicles, which is done using innoextract, and it's deselected at compile time. Could you produce a new build with innoextract compiled in or just change the existing build? Thanks in advance!

**Comment by dr460nf1r3** [comment]
What kind of adjustments do we need in the PKGBUILD? Please create an issue at the Chaotic packages repo so we can discuss it. Thanks!
